### PR TITLE
 Implement is_lose in game.py

### LINKED ISSUE #12: Implement check_if_lose in game.py

### PR DESCRIPTION
**Pull request by EdwardHaoranLee**

### CODE REVIEW

File: Woodoku/src/game.py [hunk 1]
```diff
@@ -24,8 +24,12 @@ def random_shapes(shapes: List[WoodokuShape], num: int) -> List[WoodokuShape]:
     pass
 
 
-def check_if_lose(board: WoodokuBoard, shapes: List[WoodokuShape], shape_availability: List[bool]) -> bool:
-    pass
+def is_lose(board: WoodokuBoard, shapes: List[WoodokuShape], shape_availability: List[bool]) -> bool:
```
Comment (Xiaorui-Huang) [inline Woodoku/src/game.py:27]:
```suggestion
def is_out_of_space(board: WoodokuBoard, shapes: List[WoodokuShape], shape_availability: List[bool]) -> bool:
```

File: Woodoku/src/game.py [hunk 2]
```diff
@@ -49,7 +53,7 @@ def game(ui: UIInterface) -> None:
         # When there is any shape in this round still waiting to be chosen.
         while any(shape_availability):
 
-            if check_if_lose(board, shapes, shape_availability):
+            if is_lose(board, shapes, shape_availability):
```
Comment (Xiaorui-Huang) [inline Woodoku/src/game.py]:
```suggestion
            if is_out_of_space(board, shapes, shape_availability):
```

Comment (EdwardHaoranLee) [reply Woodoku/src/game.py]: Yeah this is a better name.

File: Woodoku/src/game.py [hunk 3]
```diff
@@ -24,8 +24,12 @@ def random_shapes(shapes: List[WoodokuShape], num: int) -> List[WoodokuShape]:
     pass
 
 
-def check_if_lose(board: WoodokuBoard, shapes: List[WoodokuShape], shape_availability: List[bool]) -> bool:
-    pass
+def is_lose(board: WoodokuBoard, shapes: List[WoodokuShape], shape_availability: List[bool]) -> bool:
+    for i in range(len(shapes)):
+        if shape_availability[i] and board.can_add_shape_to_board(shapes[i]):
+            return True
```
Comment (Xiaorui-Huang) [inline Woodoku/src/game.py]:
```suggestion
def is_out_of_space(board: WoodokuBoard, shapes: List[WoodokuShape], shape_availability: List[bool]) -> bool:
    for i, shape in enumerate(shapes):
        if shape_availability[i] and board.can_add_shape_to_board(shape):
            return True
```

Comment (EdwardHaoranLee) [reply Woodoku/src/game.py]: Cool this is nicer.

**Review by Xiaorui-Huang — APPROVED** [review]

**Pull request merged**